app/org/sagebionetworks/bridge/cache/CacheProvider.java

```diff
@@ -164,7 +165,14 @@ public UserSession getUserSession(final String sessionToken) {
             if (ser == null) {
                 return null;
             }
-            return bridgeObjectMapper.readValue(ser, UserSession.class);
+            try {
+                return bridgeObjectMapper.readValue(ser, UserSession.class);
+            } catch (IOException ex) {
+                // Because StudyParticipant.Builder.withEncryptedHealthCode() can throw, and because of the way
+                // Jackson's BuilderBasedDeserializer, the error message actually contains the JSON. To prevent leaking
+                // personal identifying info, we should squelch the error message and replace it with our own.
+                throw new BridgeServiceException("Error parsing JSON for session " + sessionToken);
+            }
```

Comment:
I wonder if we can configure BridgeObjectMapper to do this, and prevent leaking of JSON parsing errors into the logs everywhere.